packages/core/upload/server/services/upload.js

```diff
@@ -114,6 +115,38 @@ module.exports = ({ strapi }) => ({
     return entity;
   },
 
+  async enhanceAndValidateBuffer(buffer, fileInfo = {}, metas = {}) {
+    const currentFile = await this.formatFileInfo(
+      {
+        filename: fileInfo.name,
+        type: fileInfo.type,
+        size: Number(buffer.length),
```

Comment:
Where is this data coming from. The issue with buffers is they don't really contain any meta data about the file such as file name and such and you have to infer the type. Also I think Strapi wants size in kb not bytes which is what buffer.length would return.

Reply:
`formatFileInfo` takes care of the conversion.